### ISSUE TITLE
Error when processing payments in currencies not eligible for India subscription mandates

### ISSUE THREAD

**Issue by james-allan** [woocommerce/woocommerce-gateway-stripe#2988]
**Describe the bug**

We've receive at least 1 merchant reporting this error when processing subscription related payments. 

<p align="center">
<img width="400" alt="Screenshot 2024-03-13 at 5 02 31 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/0bb98066-99bb-4866-ab97-b5a49c6c8c65">
</p>

The error is caused by the Stripe plugin including mandate data necessary to process subscription payments from India issued cards. Stripe docs ([direct link](https://docs.stripe.com/india-recurring-payments?integration=paymentIntents-setupIntents#create-mandate)) suggest that: 

> **Note**
> You can pass the payment_method_options[card][mandate_options] parameter for all requests. Stripe ignores these parameters if your customer is using a non-India issued card because the regulation doesn’t apply to them.

However it appears there may be some validation of the transaction currency. I haven't been able to replicate this issue on a test account using an invalid currency (AUD, MXN) and an India issued card. 

**Steps to test**

1. Set your store currency to an in-eligible currency. eg MXN.
2. Install Woo Subscriptions. 
3. Create a subscription product if you don't have one. 
4. Purchase the subscription using a card issued in India. 
   - Stripe provide these test cards: https://docs.stripe.com/india-recurring-payments?integration=paymentIntents-setupIntents#testing
   - `4000003560000008` is another generic India card you can use for testing.
5. According to this error a merchant has reported. This should lead to an error. 

I cannot replicate myself. It could be a test vs live mode thing.

**Expected behavior**
Based on the error in Stripe, we could exclude this mandate data when the order currency is an invalid currency based on the list found here: https://docs.stripe.com/india-recurring-payments?integration=paymentIntents-setupIntents#create-mandate

**Additional context**

7879498-zd-a8c (p1710300612177349-slack-C7U3Y3VMY)

**Comment by EjayhanFernandes** [comment]
Experienced in 7867212-zd-a8c -  Currency is AUD. They mentioned that this problem arises only when Stripe is in Test Mode and they are using a provided card.

**Comment by EjayhanFernandes** [comment]
Experienced in 7851708-zd-a8c - Currency is PLN.

**Comment by EjayhanFernandes** [comment]
Experienced in 7819134-zd-a8c - Currency is HUF.

**Comment by maxrice** [comment]
This also occurs when processing a subscription renewal payment for an imported subscription where a mandate wasn't previously created:
> Mandate or subscription cannot have start date before yesterday in IST. Expected start_date to be at or after 2024-03-12 18:30 UTC (1710268200) but was 2023-03-13 12:38 UTC (1678711104).

Stripe appears to be doing validation of all mandate data even when it's supposed to be ignoring that attribute.

**Comment by james-allan** [comment]
Thanks for the reports. Given this affecting a number of merchants, I've bumped the priority to high. 

> This also occurs when processing a subscription renewal payment for an imported subscription where a mandate wasn't previously created:

@maxrice yeah I believe that would be expected given the circumstances. As part of v7.8 of Stripe we fixed an issue where subscriptions which didn't already have a mandate, would have one created on their next renewal. 

The issue there is that the start date which could be well in the past fails Stripe's validation. We'll need to get that fixed up for the case you mentioned where a mandate is generated after the subscription started. There's an open issue for that here: https://github.com/woocommerce/woocommerce-gateway-stripe/issues/2946

> Stripe appears to be doing validation of all mandate data even when it's supposed to be ignoring that attribute.

Yeah I suspect something changed on their end because this mandate to fix recurring charges using cards issued in India has been around for a while now and only just started resulting in failures recently from what I can tell.
